tokens/usecase/tokens_usecase.go

```diff
@@ -349,26 +391,69 @@ func (t *tokensUseCase) getPricesForBaseDenom(ctx context.Context, baseDenom str
 }
 
 func (t *tokensUseCase) getChainScalingFactorMut(precision int) (osmomath.Dec, bool) {
-	result, ok := t.precisionScalingFactorMap[precision]
-	return result, ok
+	result, ok := t.precisionScalingFactorMap.Load(precision)
+	if !ok {
+		return osmomath.Dec{}, false
+	}
+
+	v, ok := result.(osmomath.Dec)
+	if !ok {
+		return osmomath.Dec{}, false
+	}
+
+	return v, true
+}
+
+// UpdateAssetsAtHeightInterval updates assets at configured height interval.
+// Internally, it calls LoadTokensFromChainRegistry as a goroutine.
+func (t *tokensUseCase) UpdateAssetsAtHeightInterval(height uint64) {
```

Comment:
Should we cover with tests this method? Any suggestions how we can do that?

Reply:
Yes. For the context of this method, we simply need to validate the following:
- `LoadTokensFromChainRegistry` is called at the expected height interval
- `LoadTokensFromChainRegistry` is not called at the expected height interval

Other invariants are to be validated by unit tests of the downstream methods.

We might have to rely on mocks to properly set this up.

Reply:
Thanks, I have added test coverage for this method!